components/builder-api/src/server/resources/origins.rs

```diff
@@ -1158,14 +1202,121 @@ fn list_origin_invitations(req: HttpRequest,
     }
 }
 
+#[allow(clippy::needless_pass_by_value)]
+fn get_origin_member_role(req: HttpRequest,
+                          path: Path<(String, String)>,
+                          state: Data<AppState>)
+                          -> HttpResponse {
+    let (origin, username) = path.into_inner();
+
+    if let Err(err) = authorize_session(&req, Some(&origin), RoleAction::Read, None) {
+        return err.into();
+    }
+
+    let conn = match state.db.get_conn().map_err(Error::DbError) {
+        Ok(conn_ref) => conn_ref,
+        Err(err) => return err.into(),
+    };
+
+    // The account id of the user being requested
+    let (target_user_id, _) = match Account::get(&username, &*conn).map_err(Error::DieselError) {
+        Ok(account) => (account.id, account.name),
+        Err(err) => {
+            debug!("{}", err);
+            return err.into();
+        }
+    };
+
+    match OriginMember::member_role(&origin, target_user_id, &*conn) {
```

Comment:
I'm not sure I follow. In the case of this function specifically you should not be able to get the role of a member if you are also not a member of the origin. Does that answer your question?

Reply:
My question is more "Can I, as a member of origin O, request the role of Person in O, when Person is not a member of O" or in more concrete terms;  if we removed @smacfarlane  from the `habitat` origin, could you then make a query "What is @smacfarlane role in the `habitat` origin?".  

I'm not sure I see that case handled, unless it's in the `Err` arm of the `member_role` query. This may just be my unfamiliarity with Diesel.

Reply:
Correct, they cannot. In this case all auth is keyed to be origin specific as it was before. So in this case if you are not a member of an origin you cannot read the roles of users in that origin. I've just added a test case for this so that this is explicit!